File: CS/WindowsApplication12/Form1.cs
using System;
using System.Windows.Forms;
// ...

namespace WindowsApplication12 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport1 detailReport = new XtraReport1();
            detailReport.CreateDocument();

            XtraReport2 masterReport = new XtraReport2();
            masterReport.CreateDocument();
            
            masterReport.PrintingSystem.Pages.RemoveAt(detailReport.currentPage);
            masterReport.ShowPreview(); 
        }

    }
}

File: VB/WindowsApplication12/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
' ...

Namespace WindowsApplication12
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim detailReport As New XtraReport1()
			detailReport.CreateDocument()

			Dim masterReport As New XtraReport2()
			masterReport.CreateDocument()
			masterReport.PrintingSystem.Pages.RemoveAt(detailReport.currentPage)

			Dim printTool As New ReportPrintTool(masterReport)
			printTool.ShowPreview()
		End Sub

	End Class
End Namespace